code/functions/bus_stop_census_code_functions.R

```diff
@@ -0,0 +1,104 @@
+    
+    routes <- fread(here("data","raw_data", "gtfs", gtfs_date, "routes.txt"), 
+                             stringsAsFactors = FALSE, 
+                             colClasses = c(route_id = "character",
+                                            route_short_name = "character",
+                                            route_long_name = "character",
+                                            route_type = "integer"),
+                             select = c("route_id","route_short_name","route_long_name","route_type"))
+    
+    stop_times <- fread(here("data","raw_data", "gtfs", gtfs_date, "stop_times.txt"),
+                                 stringsAsFactors = FALSE, 
+                                 colClasses = c("character","character","character","character","integer"),
+                                 select = c("trip_id","arrival_time","stop_id","stop_sequence"))
+    
+    shapes <- fread(here("data","raw_data", "gtfs", gtfs_date, "shapes.txt"), 
+                             stringsAsFactors = FALSE, 
+                             colClasses = c("character","double","double","integer"))
+    
+    trips <- fread(here("data","raw_data", "gtfs", gtfs_date, "trips.txt"),
+                            stringsAsFactors = FALSE,
+                            colClasses = c("character","integer","character",
+                                           "character","integer","character",
+                                           "character"),
+                            drop = c("block_id"))
+    
+    calendar <- fread(here("data","raw_data", "gtfs", gtfs_date, "calendar.txt"), 
+                               stringsAsFactors = FALSE, 
+                               colClasses = c("integer","character","character",
+                                              "character","character","character",
+                                              "character","character","character",
+                                              "character"),
+                               select = c("service_id"))
+    
+    
+    ## CLEAN GTFS DATA --------------------------------------------------------------------------------------------------
+    # Convert service_id to day of the week text
+    calendar <- calendar %>% mutate(.,service_type = 
+                                                        with(.,case_when((service_id == 3) ~ "Saturday",
+                                                                         (service_id == 4) ~ "Sunday",
+                                                                         (service_id == 5) ~ "Weekday")))
+    # Convert route_type to bus or rail
+    routes <- routes %>% mutate(.,route_type_desc = 
+                                                    with(.,case_when((route_type == 3) ~ "Bus",
+                                                                     (route_type == 1) ~ "Rail")))
+    
+    # Format Stop Times To be 24 hour
```

Comment:
This comments should probably be something like "Convert stop hours that are into the next day to regular clock hours (e.g. 25:12:05 becomes 01:12:05)"